ros2topic/ros2topic/verb/info.py

```diff
@@ -12,19 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSReliabilityPolicy
+
 from ros2cli.node.direct import DirectNode
 from ros2topic.api import get_topic_names_and_types
 from ros2topic.api import TopicNameCompleter
 from ros2topic.verb import VerbExtension
 
 
+def print_topic_endpoint_info(topic_name, get_topic_endpoint_info_func):
+    for info in get_topic_endpoint_info_func(topic_name):
+        print('Node name: %s' % info['node_name'])
+        print('Node namespace: %s' % info['node_namespace'])
+        print('Topic type: %s' % info['topic_type'])
+        print('GID: %s' % '.'.join(format(x, '02x') for x in info['gid']))
+        print('QoS profile:')
+        qos_profile = info['qos_profile']
+        print('  Reliability: %s' % QoSReliabilityPolicy(qos_profile['reliability']).name)
+        print('  Durability: %s' % QoSDurabilityPolicy(qos_profile['durability']).name)
+        print('  Lifespan: %d nanoseconds' % qos_profile['lifespan'].nanoseconds)
+        print('  Deadline: %d nanoseconds' % qos_profile['deadline'].nanoseconds)
+        print('  Liveliness: %s' % QoSLivelinessPolicy(qos_profile['liveliness']).name)
+        print('  Liveliness lease duration: %d nanoseconds\n' %
+              qos_profile['liveliness_lease_duration'].nanoseconds)
+
 class InfoVerb(VerbExtension):
     """Print information about a topic."""
 
     def add_arguments(self, parser, cli_name):
         arg = parser.add_argument(
             'topic_name',
             help="Name of the ROS topic to get info (e.g. '/chatter')")
+        parser.add_argument(
+            '--verbose', '-v', action='store_true',
+            help='Prints detailed information like the node name, node namespace, topic type, '
+                 'GUID and QoS Profile of the publishers and subscribers to this topic')
```

Comment:
nitpick: weird indention again

Reply:
I think these weird indentations are for satisfying `flake8`.

Reply:
That's surprising to me, since we never usually have indentations that are not a multiple of four and usually never anything other than exactly four. I guess if the linter is happy though...